### PR TITLE
Redirect mobile clients to warning page

### PR DESCRIPTION
**Pull request by bvaughn**
cc @jonbell-lot23 if you want to tweak the UI of this page

### CODE REVIEW

**Review by jonbell-lot23 — APPROVED** [review]

**Comment by jonbell-lot23** [comment]
I notice that this screen appears after the login page, rather than before. It's better to make it before. Failing fast is better than "Please go through the log in process, oops, actually the site won't look right, but we didn't tell you first."

**Comment by bvaughn** [comment]
> I notice that this screen appears after the login page, rather than before. It's better to make it before. Failing fast is better than "Please go through the log in process, oops, actually the site won't look right, but we didn't tell you first."

That's a good catch. I am not sure how our auth is configured. I'll have to look into it and see if I can override it per-route like this.

File: middleware.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export function middleware(request: NextRequest) {
+  const { ua } = userAgent(request);
+
+  if (isMobile(ua)) {
+    // If the user is attempting to visit Replay on a mobile device for the first time,
+    // show them a message that it has not been optimized for mobile
+    // If they have already confirmed this message (detectable via a cookie) then let them through
+    const cookie = request.cookies.get("mobile-warning-dismissed");
+    if (cookie == null && request.nextUrl.pathname !== "/mobile-warning") {
```
Comment (bvaughn) [inline middleware.ts:12]: The `pathname` check is probably unnecessary, since the middleware doesn't respond to that route.

**Review by hbenl — APPROVED** [review]

**Comment by bvaughn** [comment]
I see an e2e test failing on this branch that I am pretty sure is unrelated to this change. I filed FE-2346 to follow up.

**Pull request merged**